discretisedfield/tests/test_field.py

```diff
@@ -3029,6 +3029,22 @@ def test_mpl_lightess(test_field):
     plt.close("all")
 
 
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+def test_mpl_dimension_lightness(valid_mesh, nvdim):
+    field = df.Field(valid_mesh, nvdim=nvdim)
+
+    if valid_mesh.region.ndim != 2:
+        with pytest.raises(RuntimeError):
+            field.mpl.lightness()
+    elif nvdim not in [1, 2]:
+        with pytest.raises(ValueError):
+            field.mpl.lightness()
```

Comment:
Lightness also works for `nvdim=3`. Both 2 and 3 vdims require `vdim_mapping` to be defined. I think we should modify the implementation and raise a `RuntimeError` when `nvdim > 3`.